### DISCUSSION TITLE
How do I use Discord.ui

### DISCUSSION THREAD

**Discussion by TriangularDev** [Rapptz/discord.py#8238] [Q&A] [answered]
I heard discord.py has button, dropdown, and slash command support. Is there any documentation for this as I can't find it anywhere.

**Comment by trevorflahardy** [comment, marked as answer]
You heard correct! Discord.py has support for buttons, dropdowns, and application commands! All these features are on the master branch of the library, they aren't on the stable version of discord.py.

### How do I install the master branch?
#### Windows
- No voice support: `py -m pip install -U git+https://github.com/Rapptz/discord.py`
- Or with voice support: `py -m pip install -U git+https://github.com/Rapptz/discord.py#egg=discord.py[voice]`
#### Linux/Mac
- No voice support: `python3 -m pip install -U git+https://github.com/Rapptz/discord.py`
- Or with voice support: `python3 -m pip install -U git+https://github.com/Rapptz/discord.py#egg=discord.py[voice]`

### How do I create a button or dropdown?
Check out the [examples on how Views work](https://github.com/Rapptz/discord.py/tree/master/examples/views) for some more info on that.

### How do I create a slash command?
Check out [this example](https://github.com/Rapptz/discord.py/blob/master/examples/app_commands/basic.py) on slash commands. In addition to this, feel free to check out this [gist which describes application commands in more detail](https://gist.github.com/AbstractUmbra/a9c188797ae194e592efe05fa129c57f).

### Documentation?
Documentation for the master branch can be found [here](https://discordpy.readthedocs.io/en/latest/index.html), and support for migration can be found [here](https://discordpy.readthedocs.io/en/latest/migrating.html). It will be very beneficial if you joined the [discord.py Discord server](https://discord.gg/r3sSKJJ) as you can get **on demand help** for the library!

**Reply by TriangularDev** [reply]
Thank you! I thought discord.py stopped development.

**Reply by trevorflahardy** [reply]
> Thank you! I thought discord.py stopped development.

For about 6 months or so Danny halted development on the library, but he's back!

`[2021-8-28]` Danny stops dpy development: https://gist.github.com/Rapptz/4a2f62751b9600a31a0d3c78100287f1
`[2022-3-06]` Danny starts dpy development again, after announcement of v10: https://gist.github.com/Rapptz/c4324f17a80c94776832430007ad40e6
`[2022-3-14]` Danny reworks async stuff, breaking changes for all! https://gist.github.com/Rapptz/6706e1c8f23ac27c98cee4dd985c8120

**Reply by aurkaxi** [reply]
how do i know which version is installed on my device?

**Reply by HigherOrderLogic** [reply]
`pip show discord.py`

**Reply by aurkaxi** [reply]
that shows '2.0.0'. is that stable branch or master branch?

**Reply by trevorflahardy** [reply]
> that shows '2.0.0'. is that a stable branch or a master branch?

2.0 is now the stable version of discord.py (can be installed with `pip install discord.py`).
